### PR TITLE
Add a few advanced activation layers

### PR DESCRIPTION
**Pull request by caisq**
* LeakyReLU
* ELU
* ThresholdedReLU
* Softmax

Fixes: https://github.com/tensorflow/tfjs-converter/issues/65

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-layers/106)

### CODE REVIEW

**Comment by nsthorat** [comment]
:lgtm_strong:

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---

*[src/exports.ts, line 197 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/106#-L96X7Fr4IMDkgqkGCUp:-L96X7Fr4IMDkgqkGCUq:b7gehi8) ([raw file](https://github.com/tensorflow/tfjs-layers/blob/8366396c17f3b9e762111891976113b91d67e423/src/exports.ts#L197)):*
> ```XML
>   @doc({
>     heading: 'Layers',
>     subheading: 'Advanced Activations',
> ```

Can you also send a PR to add this to skeleton.json in website so we pin this subheading?

---

*[src/layers/advanced_activations.ts, line 36 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/106#-L96WZU14lwawXo0AgrM:-L96WZU14lwawXo0AgrN:bepf509) ([raw file](https://github.com/tensorflow/tfjs-layers/blob/8366396c17f3b9e762111891976113b91d67e423/src/layers/advanced_activations.ts#L36)):*
> ```XML
>  *
>  * It allows a small gradient when the unit is not active:
>  * `f(x) = alpha *x for x < 0.`
> ```

space after * 

---

*[src/layers/advanced_activations.ts, line 157 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/106#-L96WpI416KUhNYDoV2p:-L96WpI416KUhNYDoV2q:bocxowf) ([raw file](https://github.com/tensorflow/tfjs-layers/blob/8366396c17f3b9e762111891976113b91d67e423/src/layers/advanced_activations.ts#L157)):*
> ```XML
>  * It follows:
>  * `f(x) = x for x > theta`,
>  * `f(x) = 0 ohterwise`.
> ```

otherwise

---

*[src/layers/advanced_activations.ts, line 170 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/106#-L96VhqM6-5NyI0fQFbF:-L96VhqM6-5NyI0fQFbG:b-v5u2nl) ([raw file](https://github.com/tensorflow/tfjs-layers/blob/8366396c17f3b9e762111891976113b91d67e423/src/layers/advanced_activations.ts#L170)):*
> ```XML
>  * Features](http://arxiv.org/abs/1402.3337)
>  */
> export class ThresholdedReLU extends Layer {
> ```

for the future, it may be nice to have "advanced_activations" be a directory, and "ThresholdedReLU" be a file inside of it. Having multiple classes in a single file I think can get a little unwieldy

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/106#-:-L96XFPG6IPTy6n4DwZv:b-jbsf8z)*

**Comment by caisq** [comment]
Review status: 0 of 3 files reviewed at latest revision, 4 unresolved discussions, all commit checks successful.

---

*[src/exports.ts, line 197 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/106#-L96X7Fr4IMDkgqkGCUp:-L96akbi4fGRgvyItGgi:b3e23z6) ([raw file](https://github.com/tensorflow/tfjs-layers/blob/8366396c17f3b9e762111891976113b91d67e423/src/exports.ts#L197)):*
<details><summary><i>Previously, nsthorat (Nikhil Thorat) wrote…</i></summary><blockquote>

Can you also send a PR to add this to skeleton.json in website so we pin this subheading?
</blockquote></details>

Done: https://github.com/tensorflow/tfjs-website/pull/111

---

*[src/layers/advanced_activations.ts, line 36 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/106#-L96WZU14lwawXo0AgrM:-L96an6-2kxGXUO2aKMt:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-layers/blob/8366396c17f3b9e762111891976113b91d67e423/src/layers/advanced_activations.ts#L36)):*
<details><summary><i>Previously, nsthorat (Nikhil Thorat) wrote…</i></summary><blockquote>

space after * 
</blockquote></details>

Done.

---

*[src/layers/advanced_activations.ts, line 157 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/106#-L96WpI416KUhNYDoV2p:-L96apOZ0GX9FOOYsyOn:b-896fix) ([raw file](https://github.com/tensorflow/tfjs-layers/blob/8366396c17f3b9e762111891976113b91d67e423/src/layers/advanced_activations.ts#L157)):*
<details><summary><i>Previously, nsthorat (Nikhil Thorat) wrote…</i></summary><blockquote>

otherwise
</blockquote></details>

Done.

---

*[src/layers/advanced_activations.ts, line 170 at r1](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/106#-L96VhqM6-5NyI0fQFbF:-L96apzQ4hFPimrK60qz:b-3mpu6a) ([raw file](https://github.com/tensorflow/tfjs-layers/blob/8366396c17f3b9e762111891976113b91d67e423/src/layers/advanced_activations.ts#L170)):*
<details><summary><i>Previously, nsthorat (Nikhil Thorat) wrote…</i></summary><blockquote>

for the future, it may be nice to have "advanced_activations" be a directory, and "ThresholdedReLU" be a file inside of it. Having multiple classes in a single file I think can get a little unwieldy
</blockquote></details>

Let's discuss this. I actually think putting multiple classes in a file is nicer than having many files.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/tensorflow/tfjs-layers/106)*
